### PR TITLE
Introduce a simplified version of td3 with warrowing (see PR 1611)

### PR DESCRIPTION
**Pull request by FelixKrayer**
Copied from PR #1611 by @arkocal 
The simplified warrowing solver

As to which solver we should keep as the simplified version, this one:

1. Can be documented a little more for didactic purposes, but I find it to be very readable as is.
2. We should find the paper most closely matching this solver.
3. This one does not pass all tests due to narrowing.

I created this pullrequest so I can address comments and do changes.
Addressing comments by @sim642 :
1. removed `remove-wpoint` option. Having this option is not necessary for this solver. However, I changed it to always remove wpoints, without checking the config, as this is the default setting in the config.
2. solver data is no longer a record. There is no need for that
3. moved query/side out. There is no real reason they were inside iterate, it was just the way it turned out after deleting  
4. TODO: decide if phases should be included. This solver does not have them, since the parallel solvers do not use them and it was created for comparison with those. However, for generalizing this solver I see the use of implementing the phases.

### CODE REVIEW

**Comment by michael-schwarz** [comment]
@sim642 @arkocal: What is you take on phases or not?
I don't have a strong opinion here, but think we should move forward with merging this.

**Comment by arkocal** [comment]
I am fine with either variant, so I would have a light preference to just proceed and merge. @sim642 had the point that the basic example should have a termination guarantee, which is reasonable, just not a deal breaker for me as I find warrowing a little more intuitive than phases.

tldr: I would keep it as is, but have no strong preference, so Simmo please feel free to override this if you do.

File: src/solver/td_simplified.ml
Comment (sim642) [inline src/solver/td_simplified.ml:1]: This module needs to be added to `Goblint_lib` to fix the CI failure.

Also, this PR should be adapted to #1605, i.e., the new solver added to the cram test along with all the others.

**Review by sim642 — CHANGES_REQUESTED** [review]

**Review by sim642 — APPROVED** [review]
